src/Tasks/Microsoft.NET.Build.Tasks/GenerateShims.cs

```diff
@@ -0,0 +1,173 @@
+        public string ProjectAssetsFile { get; set; }
+
+        /// <summary>
+        /// The file name of Apphost asset.
+        /// </summary>
+        [Required]
+        public string DotNetAppHostExecutableNameWithoutExtension { get; set; }
+
+        /// <summary>
+        /// Path to project file (.csproj|.vbproj|.fsproj)
+        /// </summary>
+        [Required]
+        public string ProjectPath { get; set; }
+
+        /// <summary>
+        /// TFM to use for compile-time assets.
+        /// </summary>
+        [Required]
+        public string TargetFrameworkMoniker { get; set; }
+
+        /// <summary>
+        /// PackageId of the dotnet tool NuGet Package.
+        /// </summary>
+        [Required]
+        public string PackageId { get; set; }
+
+        /// <summary>
+        /// Package Version of the dotnet tool NuGet Package.
+        /// </summary>
+        [Required]
+        public string PackageVersion { get; set; }
+
+        /// <summary>
+        /// The command name of the dotnet tool.
+        /// </summary>
+        [Required]
+        public string ToolCommandName { get; set; }
+
+        /// <summary>
+        /// The entry point of the dotnet tool which will be run by Apphost
+        /// </summary>
+        [Required]
+        public string ToolEntryPoint { get; set; }
+
+        /// <summary>
+        /// The output Directroy path of generated shims.
+        /// </summary>
+        [Required]
+        public string PackagedShimOutputDirectory { get; set; }
+
+        /// <summary>
+        /// The RuntimeIdentifiers that shims will be generated for.
+        /// </summary>
+        [Required]
+        public ITaskItem[] PackageToolShimRuntimeIdentifiers { get; set; }
+
+        /// <summary>
+        /// Path of generated shims. metadata "ShimRuntimeIdentifier" is used to map back to PackageToolShimRuntimeIdentifiers.
+        /// </summary>
+        [Output]
+        public ITaskItem[] EmbeddedApphostPaths { get; private set; }
+
+        protected override void ExecuteCore()
+        {
+            NuGetFramework targetFramework = NuGetUtils.ParseFrameworkName(TargetFrameworkMoniker);
+            LockFile lockFile = new LockFileCache(this).GetLockFile(ProjectAssetsFile);
+            _packageResolver = NuGetPackageResolver.CreateResolver(lockFile, ProjectPath);
+
+            var embeddedApphostPaths = new List<ITaskItem>();
+            foreach (var runtimeIdentifier in PackageToolShimRuntimeIdentifiers.Select(r => r.ItemSpec))
+            {
+                var resolvedApphostAssetPath = GetApphostAsset(targetFramework, lockFile, runtimeIdentifier);
+
+                var packagedShimOutputDirectoryAndRid = Path.Combine(PackagedShimOutputDirectory, runtimeIdentifier);
+                var appHostDestinationFilePath =
+                    Path.Combine(packagedShimOutputDirectoryAndRid,
+                        $"{ToolCommandName}{Path.GetExtension(resolvedApphostAssetPath)}");
```

Comment:
nit: I find `ToolCommandName + Path.GetExtension(resolvedApphostAssetPath)` easier to read. It's also faster than string.Format. but perf isn't really my concern here.

Reply:
fixed, changed as suggested